### ISSUE TITLE
Weights are reset to zero when changes occur in the mesh.

### ISSUE THREAD

**Issue by MarkMironov** [umiyuki/Artstage-Issues#87]
https://youtu.be/JGghgVyChz8

The video shows how I set the weight of all the edges to 5. Then I add new edges and immediately delete them, and we see that the weights are reset.
Weight of each edge should be unchanged. Otherwise, what's the point in weights if any change in the mesh can nullify them?

**Comment by umiyuki** [comment]
Edited vertices and edge weights are deleted. Because when a vertex or edge is edited, they are internally deleted once and a new one is inserted.
This is not a bug, it is the specification now. It may be improved in the future.
Please set weights after finish editing the mesh.

**Comment by MarkMironov** [comment]
@umiyuki 
Ок. I ask you not to remove this issue, since improvement is really necessary.
Imagine a situation if an artist painted a landscape, and then noticed that clear strokes were blurry, and now he would have to add detail again. And then again he discovers that the strokes are blurry. This is exactly what happens with the mesh. This annoys and kills the urge to modeling using a subdiv.

**Comment by umiyuki** [comment]
It's a shame to get you frustrated while modeling, but anyway it's difficult to modeling with the setting smooth weights in same time in Artstage now.
I assume that smooth weight paint is used to fine-tune the smooth at the end of the work.
After all, If this tool didn't exist you wouldn't even get frustrated. You said it should be better exist than no exist about sculpting, but about smooth weight paint you're frustrated with half-finished features.

**Comment by MarkMironov** [comment]
@umiyuki 
Thanks to Artstage, I found out that I can give edges weight. After that I began to use weights even in other programs. Now for me it's a familiar tool without which I can't do.

No, I don't use weights at the end of the work, but from beginning to end. Since I always use subdivs from the very beginning and without weights, shapes are too rounded.

**Comment by umiyuki** [comment]
In ver0.9.1, the weight is maintained even when using the knife tool and the edge loop insertion tool.

**Comment by MarkMironov** [comment]
@umiyuki 
This is not true. On the video pay attention to the center line of the leaf. Using a knife, I constantly had to reassign her weight.
https://youtu.be/pbtapsyym_s

**Comment by umiyuki** [comment]
I tried, but the problem doesn't reappear.
And no one said that the weight is maintained by the delete tool.

**Comment by MarkMironov** [comment]
@umiyuki I just tried again, without using delete tool. I just created a leaf, set the weight in the center line of the leaf and along its edge, then cut it like in the video. Weight of the center line has been reset to zero, and has been preserved along the edge.

**Comment by umiyuki** [comment]
Please submit a file that reproduces the problem.
https://www.dropbox.com/request/HOU04JVUNXAlPxqmOMUu
